### PR TITLE
Deploy config containing secrets as Secret

### PR DESCRIPTION
**Pull request by kvanzuijlen**
Added support for deploying the config.json as a secret

### CODE REVIEW

File: charts/renovate/templates/configmap.yaml [hunk 1]
```diff
@@ -1,11 +1,17 @@
 {{- if not .Values.renovate.existingConfigFile }}
 apiVersion: v1
-kind: ConfigMap
 metadata:
   name: {{ template "renovate.fullname" . }}-config
   labels:
     {{- include "renovate.labels" . | nindent 4 }}
+{{- if .Values.renovate.configIsSecret }}
```
Comment (viceice) [inline charts/renovate/templates/configmap.yaml:7]: please move this above the metadata

File: charts/renovate/templates/cronjob.yaml [hunk 2]
```diff
@@ -76,12 +76,12 @@ spec:
               {{- if or .Values.dind.enabled .Values.cronjob.preCommand}}
               command: ["/bin/bash", "-c"]
               args:
-                - | 
-                {{- if .Values.dind.enabled }} 
+                - |
```
Comment (viceice) [inline charts/renovate/templates/cronjob.yaml:79]: avoid unrelated changed

Comment (kvanzuijlen) [reply charts/renovate/templates/cronjob.yaml:79]: Sorry, triggerhappy IDE here, I'll roll back these changes!

**Comment by kvanzuijlen** [comment]
@viceice Do you agree with conditionally creating a secret/configmap in the configmap.yaml? I think it keeps it better maintainable (no duplications) but it might be less readable.

File: charts/renovate/templates/configmap.yaml (hunk 1 above)
Comment (kvanzuijlen) [reply charts/renovate/templates/configmap.yaml:3]: ✅

**Comment by viceice** [comment]
> @viceice Do you agree with conditionally creating a secret/configmap in the configmap.yaml? I think it keeps it better maintainable (no duplications) but it might be less readable.

we should probably rename the file to some more generic name

**Comment by kvanzuijlen** [comment]
> > @viceice Do you agree with conditionally creating a secret/configmap in the configmap.yaml? I think it keeps it better maintainable (no duplications) but it might be less readable.
> 
> we should probably rename the file to some more generic name

My proposal would be to just call it `config.yaml`

**Comment by kvanzuijlen** [comment]
@viceice I can't think of a way renaming the file (or the other changes in this PR) could be a breaking change. Do you see any potential problems?

File: charts/renovate/templates/config.yaml [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
-    {{- include "renovate.labels" . | nindent 4 }}
 data:
+{{- end }}
   config.json: |-
   {{- if not .Values.renovate.configEnableHelmTpl }}
     {{- required "A valid .Values.renovate.config entry is required if existingConfigFile is not specified!" .Values.renovate.config | nindent 4 }}
   {{- else }}
     {{- required "A valid .Values.renovate.config entry is required if existingConfigFile is not specified!" (tpl .Values.renovate.config .) | nindent 4 }}
   {{- end }}
+metadata:
```
Comment (viceice) [inline charts/renovate/templates/config.yaml:17]: please move back before data section

**Review by viceice — CHANGES_REQUESTED** [review]

File: charts/renovate/templates/config.yaml (hunk 3 above)
Comment (kvanzuijlen) [reply charts/renovate/templates/config.yaml]: Moved it back

File: charts/renovate/templates/config.yaml [hunk 4]
```diff
@@ -1,11 +1,17 @@
 {{- if not .Values.renovate.existingConfigFile }}
 apiVersion: v1
-kind: ConfigMap
 metadata:
```
Comment (viceice) [inline charts/renovate/templates/config.yaml:3]:
```suggestion
{{- if .Values.renovate.configIsSecret }}
kind: Secret
type: Opaque
{{- else }}
kind: ConfigMap
{{- end }}
metadata:
```

File: charts/renovate/templates/config.yaml [hunk 5]
```diff
@@ -1,11 +1,17 @@
...
   labels:
     {{- include "renovate.labels" . | nindent 4 }}
+{{- if .Values.renovate.configIsSecret }}
+kind: Secret
+type: Opaque
+stringData:
+{{- else }}
+kind: ConfigMap
 data:
+{{- end }}
```
Comment (viceice) [inline charts/renovate/templates/config.yaml:14]:
```suggestion
{{- if .Values.renovate.configIsSecret }}
stringData:
{{- else }}
kind: ConfigMap
data:
{{- end }}
```

**Review by viceice — COMMENTED** [review]
thise seems like a better order to mee

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
will be published with next renovate release

**Pull request merged**